### ISSUE TITLE
Onboard Scoobi to MDS

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#7276]
Onboard provider Scoobi to MDS.

We've been provided with API access and are currently testing the production API based on trips completed on 9/23.

**Comment by johnclary** [comment]
Email to Steve after testing:
> We've attempted to pull trips from the test rides, but the API is returning the same set of ~30 trips regardless of the end_time we use to query.
>  
> All of the trips returned are timestamped on Sep 27. As well, the lon/lat coordinate pairs are reversed, and upon reversing them it's clear these trips took place in Pittsburgh. 
>  
> Please advise.

And response:

> I published it and flipped the order of the lat/long in the array; it should be good now.
> 
> Could you give me an example of what your passing in for the date time val? I’m able to specify different times and it converts ok.

I replied to Steve with an example query URL we're using.

**Comment by johnclary** [comment]
I've been using this script to convert the responses sergio provided into geojson that can be inspected on geojson.io

```python
"""Creates a geojson feature collection from MDS trip. Inspect by loading in geojson.io"""
import json
from datetime import datetime

FNAME = "scoobi_sept_23_18utc"

with open(f"{FNAME}.json", "r") as fin:
  data = json.load(fin)["data"]["trips"]


trips = {'type': 'FeatureCollection', 'features':[]}

for trip in data:
  feature = {'type': 'Feature', 'properties': {}, 'geometry': {'type': 'MultiPoint', 'coordinates': []}}
  waypoints = trip["route"]["features"]
  
  # filters trips in pittsburgh
  if waypoints[0]["geometry"]["coordinates"][1] > 40:
    continue

  # mergess coordinate pairs into single feature
  for w in waypoints:
    feature["geometry"]["coordinates"].append(w["geometry"]["coordinates"])

  # asssign start and end time properties
  start = trip["start_time"]
  feature["properties"]["start_time"] = str(datetime.fromtimestamp(start/1000))

  end = trip["end_time"]
  feature["properties"]["end_time"] = str(datetime.fromtimestamp(end/1000))

  trips["features"].append(feature)


with open(f"{FNAME}.geo.json", "w") as fout:
  fout.write(json.dumps(trips))

```

**Comment by johnclary** [comment]
I notified Steve of these issues:

> We're able to query for trips and the lat/lon are now correct. However, our queries return trips in the future as well as trips that are in Pittsburgh.
> 
> As well, the sole Austin trip has three identical coordinate pairs. We would expect these to be different (assuming the vehicle moved during the trip).
> 
> Here's the query:
> `/provider/dockless/trips?end_time=2021-09-23T17`
> 
> We need:
> - Only trips in Austin
> - The query to return trips that match the requested end_time

**Comment by sergiogcx** [comment]
Thanks for the code @johnclary ! I have been doing it the lazy-man's way with Smart JSON Editor, Ctrl+C (for copy) than paste directly into geojson.io, the disadvantage being that you can only copy-paste one trip at the time:

![2021-09-28_15-52-57.png](https://images.zenhubusercontent.com/5b7edad7290aac725aec290c/3a75a8bc-d4c9-4b16-911a-7b555e3b9de1)

**Comment by johnclary** [comment]
whoa, cool editor though!

**Comment by johnclary** [comment]
TLDR: Scoobi meets our needs but is still working to meet RideReport's requirements.

> Thanks, Steve. I reviewed the Pittsburgh trips. The route features appear to be timestamped with the time of the API request. They are often identical and do not fall within the range of the start/end times. As well, the number of route points is quite low relative to the duration of the trip.
> 
> Our internal reporting relies on only the beginning and end of the route and does not reference the route feature timestamps. From our side your spec meets our needs, assuming the starting/ending route features will be accurate (ie, assuming that the issue that occurred with the Austin test trip has been resolved).
> 
> You will need to continue to work with RideReport to refine your spec as they rely on a more robust feature set.

**Comment by patrickm02L** [comment]
@johnclary Sergio walked me through turning on the DAG for Scoobi. It's live and we're monitoring it in advance of go-live on 11/1.

**Comment by johnclary** [comment]
Excellent—thanks!

**Comment by patrickm02L** [comment]
Refer to Shared Mobility dashboard to see trips. Everything else looks good.
